ddtrace/tracer.py

```diff
@@ -525,22 +499,16 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
 
             # The constant tags for the dogstatsd client needs to updated with any new
             # service(s) that may have been added.
-            self._update_dogstatsd_constant_tags()
+            if self._runtime_worker:
+                self._runtime_worker.update_runtime_tags()
 
         self._hooks.emit(self.__class__.start_span, span)
 
         return span
 
-    def _update_dogstatsd_constant_tags(self):
-        """Prepare runtime tags for ddstatsd."""
-        # DEV: ddstatsd expects tags in the form ['key1:value1', 'key2:value2', ...]
-        tags = ["{}:{}".format(k, v) for k, v in RuntimeTags()]
-        self.log.debug("Updating constant tags %s", tags)
-        self._dogstatsd_client.constant_tags = tags
-
     def _start_runtime_worker(self):
-        self._runtime_worker = RuntimeWorker(self._dogstatsd_client, self._RUNTIME_METRICS_INTERVAL)
-        self._runtime_worker.start()
+        self._runtime_worker = RuntimeWorker(self._dogstatsd_url)
+        self._runtime_metrics_started = True
```

Comment:
This doesn't seem to be used or required since we can use `self._runtime_worker.started`

Reply:
Removed, this is relevant to #1992 .